module/VuFind/tests/unit-tests/src/VuFindTest/Db/Service/SessionServiceTest.php

```diff
@@ -0,0 +1,319 @@
+            $entityManager->expects($this->exactly($count))->method('flush');
+        }
+        return $entityManager;
+    }
+
+    /**
+     * Mock queryBuilder
+     *
+     * @param string $parameter Input query parameter
+     * @param array  $result    Expected return value of getResult method.
+     *
+     * @return MockObject
+     */
+    protected function getQueryBuilder($parameter, $result)
+    {
+        $queryBuilder = $this->getMockBuilder(\Doctrine\ORM\QueryBuilder::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $queryBuilder->expects($this->once())->method('select')
+            ->with('s')
+            ->willReturn($queryBuilder);
+        $queryBuilder->expects($this->once())->method('from')
+            ->with(Session::class, 's')
+            ->willReturn($queryBuilder);
+        $queryBuilder->expects($this->once())->method('where')
+            ->with('s.sessionId = :sid')
+            ->willReturn($queryBuilder);
+        $queryBuilder->expects($this->once())->method('setParameter')
+            ->with('sid', $parameter)
+            ->willReturn($queryBuilder);
+        $query = $this->getMockBuilder(\Doctrine\ORM\AbstractQuery::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getResult'])
+            ->getMockForAbstractClass();
+        $query->expects($this->once())->method('getResult')
+            ->willReturn($result);
+        $queryBuilder->expects($this->once())->method('getQuery')
+            ->willReturn($query);
+        return $queryBuilder;
+    }
+
+    /**
+     * Session service object to test.
+     *
+     * @param MockObject      $entityManager Mock entity manager object
+     * @param MockObject      $pluginManager Mock plugin manager object
+     * @param MockObject|null $session       Mock session entity object
+     *
+     * @return MockObject
+     */
+    protected function getService(
+        $entityManager,
+        $pluginManager,
+        $session = null,
+    ) {
+        $serviceMock = $this->getMockBuilder(
+            \VuFind\Db\Service\SessionService::class
+        )
+            ->setMethods(['createEntity'])
+            ->setConstructorArgs([$entityManager, $pluginManager])
+            ->getMock();
+        if ($session) {
+            $serviceMock->expects($this->once())->method('createEntity')
+                ->willReturn($session);
+        }
+        return $serviceMock;
+    }
+
+    /**
+     * Test retriving an session object from database.
+     *
+     * @return void
+     */
+    public function testGetSessionById()
+    {
+        $session = $this->getMockBuilder(\VuFind\Db\Entity\Session::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $entityManager = $this->getEntityManager($session);
+        $pluginManager = $this->getPluginManager(true);
+        $queryBuilder = $this->getQueryBuilder('1', [$session]);
```

Comment:
You have tests for the case where a session is found and we don't want to create one, and for where a session is not found and we do want to create one. Maybe it's worth adding a third test for the case where we don't want to create a session, and an existing session is not found -- I think that code path is not yet being exercised by this test.